src/analysis/inner_product_space/l2_space.lean

```diff
@@ -387,6 +387,16 @@ begin
   exact subset_span ⟨i, rfl⟩
 end
 
+protected lemma has_sum_inner_mul_inner (b : hilbert_basis ι 𝕜 E) (x y : E) :
+  has_sum (λ i, ⟪x, b i⟫ * ⟪b i, y⟫) ⟪x, y⟫ :=
+begin
```

Comment:
Faster, I think: take `has_sum_repr` of `y`, apply the linear map `⟪x, .⟫` to it, and simp out using `repr_apply_apply`